Assimp-3.2 not recognized

Hi all,

assimp-3.2.0 is not recognized by `configure`:

```
+ cd pioneer-20160512
+ ./bootstrap
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /usr/bin/autoconf --force
autoreconf: configure.ac: not using Autoheader
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:52: installing './compile'
configure.ac:50: installing './config.guess'
configure.ac:50: installing './config.sub'
configure.ac:48: installing './install-sh'
configure.ac:48: installing './missing'
contrib/PicoDDS/Makefile.am: installing './depcomp'
parallel-tests: installing './test-driver'
autoreconf: Leaving directory `.'
+ CFLAGS='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic'
+ export CFLAGS
+ CXXFLAGS='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic'
+ export CXXFLAGS
+ FFLAGS='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -I/usr/lib64/gfortran/modules'
+ export FFLAGS
+ FCFLAGS='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -I/usr/lib64/gfortran/modules'
+ export FCFLAGS
+ LDFLAGS='-Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld'
+ export LDFLAGS
+ '[' 1 = 1 ']'
+++ dirname ./configure
++ find . -name config.guess -o -name config.sub
+ for i in '$(find $(dirname ./configure) -name config.guess -o -name config.sub)'
++ basename ./config.sub
+ '[' -f /usr/lib/rpm/redhat/config.sub ']'
+ /usr/bin/rm -f ./config.sub
++ basename ./config.sub
+ /usr/bin/cp -fv /usr/lib/rpm/redhat/config.sub ./config.sub
'/usr/lib/rpm/redhat/config.sub' -> './config.sub'
+ for i in '$(find $(dirname ./configure) -name config.guess -o -name config.sub)'
++ basename ./config.guess
+ '[' -f /usr/lib/rpm/redhat/config.guess ']'
+ /usr/bin/rm -f ./config.guess
++ basename ./config.guess
+ /usr/bin/cp -fv /usr/lib/rpm/redhat/config.guess ./config.guess
'/usr/lib/rpm/redhat/config.guess' -> './config.guess'
+ '[' 1 = 1 ']'
+ '[' x '!=' x-specs=/usr/lib/rpm/redhat/redhat-hardened-ld ']'
++ find . -name ltmain.sh
+ ./configure --build=x86_64-redhat-linux-gnu --host=x86_64-redhat-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --disable-silent-rules --with-ccache --with-strip --with-version --with-extra-version -without-extra-warnings --without-thirdparty --with-external-liblua PIONEER_DATA_DIR=/usr/share/pioneer
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... x86_64-redhat-linux-gnu
checking host system type... x86_64-redhat-linux-gnu
checking for x86_64-redhat-linux-gnu-gcc... no
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... none
checking for x86_64-redhat-linux-gnu-g++... no
checking for x86_64-redhat-linux-gnu-c++... no
checking for x86_64-redhat-linux-gnu-gpp... no
checking for x86_64-redhat-linux-gnu-aCC... no
checking for x86_64-redhat-linux-gnu-CC... no
checking for x86_64-redhat-linux-gnu-cxx... no
checking for x86_64-redhat-linux-gnu-cc++... no
checking for x86_64-redhat-linux-gnu-cl.exe... no
checking for x86_64-redhat-linux-gnu-FCC... no
checking for x86_64-redhat-linux-gnu-KCC... no
checking for x86_64-redhat-linux-gnu-RCC... no
checking for x86_64-redhat-linux-gnu-xlC_r... no
checking for x86_64-redhat-linux-gnu-xlC... no
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... none
checking for an ANSI C-conforming const... yes
checking for x86_64-redhat-linux-gnu-ranlib... no
checking for ranlib... ranlib
checking for x86_64-redhat-linux-gnu-strip... no
checking for strip... strip
checking whether make supports nested variables... (cached) yes
checking for ccache... /usr/bin/ccache
checking to see if g++ supports -std=c++11... yes
checking to see if gcc/g++ supports -Wall... yes
checking to see if gcc/g++ supports -Qunused-arguments... no
checking to see if gcc/g++ supports -fcolor-diagnostics... no
configure: using pioneer-thirdparty repository in no
checking for x86_64-redhat-linux-gnu-pkg-config... /usr/bin/x86_64-redhat-linux-gnu-pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for SIGC... yes
checking for SDL2... yes
checking for FREETYPE... yes
checking for VORBIS... yes
checking for PNG... yes
checking for ASSIMP... yes
checking for ASSIMP version >= 3.2... no
configure: error: "assimp version >= 3.2 not found. Get it from https://assimp.sourceforge.net/"
```

```
$ pkg-config --modversion assimp
3.2.0
```

@sagitter

To solve this issue, you need to look config.log file after configure command, not just a console output.
This is what config.log file shows, if I add "%global _hardened_build 1" into [pioneer.spec](http://pkgs.fedoraproject.org/cgit/rpms/pioneer.git/tree/pioneer.spec?id=e1ebe7b8973e3e3f3553a1ecdbb9557b7d9887cc) file:
```
configure:5872: checking for ASSIMP version >= 3.2
configure:5907: g++ -o conftest -I/usr/include  -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld conftest.cpp -lassimp  -ldl -lrt -lz -lassimp >&5
/usr/bin/ld: /tmp/ccyyGSh5.o: relocation R_X86_64_PC32 against undefined symbol `aiGetVersionMajor' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
configure:5907: $? = 1
configure: program exited with status 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "pioneer"
| #define PACKAGE_TARNAME "pioneer"
| #define PACKAGE_VERSION "0.00"
| #define PACKAGE_STRING "pioneer 0.00"
| #define PACKAGE_BUGREPORT "pioneer-dev@pioneerspacesim.net"
| #define PACKAGE_BUGREPORT "***@pioneerspacesim.net"
| #define PACKAGE_URL ""
| #define PACKAGE "pioneer"
| #define VERSION "0.00"
| /* end confdefs.h.  */
| #include <assimp/defs.h>
|                                    #include <assimp/version.h>
| int
| main ()
| {
| do {
|                                   if (aiGetVersionMajor() >= 3 && aiGetVersionMinor() >= 2)
|                                       return 0;
|                                   return 1;
|                                 } while(0)
|   ;
|   return 0;
| }
configure:5912: result: no
configure:5926: error: "assimp version >= 3.2 not found. Get it from https://assimp.sourceforge.net/"
```
where "***" - is some hidden email address, to not show it on public.

Therefore, to solve it, you may add "-fPIC" to ASSIMP_CFLAGS define, where your [configure stage](http://pkgs.fedoraproject.org/cgit/rpms/pioneer.git/tree/pioneer.spec?id=e1ebe7b8973e3e3f3553a1ecdbb9557b7d9887cc#n138) may look as follows:
```
%configure --disable-silent-rules --with-ccache --without-strip \
 --with-version --with-extra-version --without-extra-warnings \
 --without-thirdparty --without-external-liblua --with-no-optimise \
 PIONEER_DATA_DIR=%{_datadir}/%{name} \
 ASSIMP_CFLAGS="-I%{_includedir} -fPIC" ASSIMP_LIBS="-L%{_libdir} -lassimp"
```
It was built with following ldd result:
```
$ ldd ~/rpmbuild/BUILD/pioneer-20160512/src/pioneer 
    linux-vdso.so.1 (0x00007fffad9df000)
    libfreetype.so.6 => /lib64/libfreetype.so.6 (0x00007f25eeccb000)
    libGL.so.1 => /lib64/libGL.so.1 (0x00007f25eea5a000)
    libSDL2_image-2.0.so.0 => /lib64/libSDL2_image-2.0.so.0 (0x00007f25ee837000)
    libSDL2-2.0.so.0 => /lib64/libSDL2-2.0.so.0 (0x00007f25ee52d000)
    libsigc-2.0.so.0 => /lib64/libsigc-2.0.so.0 (0x00007f25ee325000)
    libvorbisfile.so.3 => /lib64/libvorbisfile.so.3 (0x00007f25ee11c000)
    libpng16.so.16 => /lib64/libpng16.so.16 (0x00007f25edeea000)
    libcurl.so.4 => /lib64/libcurl.so.4 (0x00007f25edc6d000)
    libassimp.so.3 => /lib64/libassimp.so.3 (0x00007f25ed41f000)
    libdl.so.2 => /lib64/libdl.so.2 (0x00007f25ed21b000)
    librt.so.1 => /lib64/librt.so.1 (0x00007f25ed012000)
    libz.so.1 => /lib64/libz.so.1 (0x00007f25ecdfc000)
    libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f25eca75000)
    libm.so.6 => /lib64/libm.so.6 (0x00007f25ec76a000)
    libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f25ec553000)
    libc.so.6 => /lib64/libc.so.6 (0x00007f25ec190000)
    libbz2.so.1 => /lib64/libbz2.so.1 (0x00007f25ebf7f000)
    libexpat.so.1 => /lib64/libexpat.so.1 (0x00007f25ebd53000)
    libxcb-dri3.so.0 => /lib64/libxcb-dri3.so.0 (0x00007f25ebb50000)
    libxcb-present.so.0 => /lib64/libxcb-present.so.0 (0x00007f25eb94c000)
    libxcb-randr.so.0 => /lib64/libxcb-randr.so.0 (0x00007f25eb73e000)
    libxcb-xfixes.so.0 => /lib64/libxcb-xfixes.so.0 (0x00007f25eb536000)
    libxcb-render.so.0 => /lib64/libxcb-render.so.0 (0x00007f25eb32b000)
    libxcb-shape.so.0 => /lib64/libxcb-shape.so.0 (0x00007f25eb127000)
    libxcb-sync.so.1 => /lib64/libxcb-sync.so.1 (0x00007f25eaf20000)
    libxshmfence.so.1 => /lib64/libxshmfence.so.1 (0x00007f25ead1c000)
    libglapi.so.0 => /lib64/libglapi.so.0 (0x00007f25eaaed000)
    libselinux.so.1 => /lib64/libselinux.so.1 (0x00007f25ea8c6000)
    libXext.so.6 => /lib64/libXext.so.6 (0x00007f25ea6b3000)
    libXdamage.so.1 => /lib64/libXdamage.so.1 (0x00007f25ea4b0000)
    libXfixes.so.3 => /lib64/libXfixes.so.3 (0x00007f25ea2aa000)
    libX11-xcb.so.1 => /lib64/libX11-xcb.so.1 (0x00007f25ea0a7000)
    libX11.so.6 => /lib64/libX11.so.6 (0x00007f25e9d67000)
    libxcb-glx.so.0 => /lib64/libxcb-glx.so.0 (0x00007f25e9b4e000)
    libxcb-dri2.so.0 => /lib64/libxcb-dri2.so.0 (0x00007f25e9948000)
    libxcb.so.1 => /lib64/libxcb.so.1 (0x00007f25e9726000)
    libXxf86vm.so.1 => /lib64/libXxf86vm.so.1 (0x00007f25e9520000)
    libdrm.so.2 => /lib64/libdrm.so.2 (0x00007f25e9310000)
    libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f25e90f4000)
    libjpeg.so.62 => /lib64/libjpeg.so.62 (0x00007f25e8e9b000)
    libtiff.so.5 => /lib64/libtiff.so.5 (0x00007f25e8c26000)
    libwebp.so.6 => /lib64/libwebp.so.6 (0x00007f25e89c8000)
    libvorbis.so.0 => /lib64/libvorbis.so.0 (0x00007f25e879c000)
    libogg.so.0 => /lib64/libogg.so.0 (0x00007f25e8594000)
    libnghttp2.so.14 => /lib64/libnghttp2.so.14 (0x00007f25e8373000)
    libidn.so.11 => /lib64/libidn.so.11 (0x00007f25e813f000)
    libssh2.so.1 => /lib64/libssh2.so.1 (0x00007f25e7f10000)
    libpsl.so.5 => /lib64/libpsl.so.5 (0x00007f25e7d03000)
    libssl3.so => /lib64/libssl3.so (0x00007f25e7abd000)
    libsmime3.so => /lib64/libsmime3.so (0x00007f25e7895000)
    libnss3.so => /lib64/libnss3.so (0x00007f25e756c000)
    libnssutil3.so => /lib64/libnssutil3.so (0x00007f25e733e000)
    libplds4.so => /lib64/libplds4.so (0x00007f25e7139000)
    libplc4.so => /lib64/libplc4.so (0x00007f25e6f34000)
    libnspr4.so => /lib64/libnspr4.so (0x00007f25e6cf5000)
    libgssapi_krb5.so.2 => /lib64/libgssapi_krb5.so.2 (0x00007f25e6aa7000)
    libkrb5.so.3 => /lib64/libkrb5.so.3 (0x00007f25e67c1000)
    libk5crypto.so.3 => /lib64/libk5crypto.so.3 (0x00007f25e6590000)
    libcom_err.so.2 => /lib64/libcom_err.so.2 (0x00007f25e638b000)
    liblber-2.4.so.2 => /lib64/liblber-2.4.so.2 (0x00007f25e617c000)
    libldap-2.4.so.2 => /lib64/libldap-2.4.so.2 (0x00007f25e5f2b000)
    libminizip.so.1 => /lib64/libminizip.so.1 (0x00007f25e5d1e000)
    libIrrXML.so.1 => /lib64/libIrrXML.so.1 (0x00007f25e5b06000)
    libpoly2tri.so.1.0 => /lib64/libpoly2tri.so.1.0 (0x00007f25e58fa000)
    /lib64/ld-linux-x86-64.so.2 (0x000055b031feb000)
    libXau.so.6 => /lib64/libXau.so.6 (0x00007f25e56f5000)
    libpcre.so.1 => /lib64/libpcre.so.1 (0x00007f25e5484000)
    libjbig.so.2.1 => /lib64/libjbig.so.2.1 (0x00007f25e5277000)
    libssl.so.10 => /lib64/libssl.so.10 (0x00007f25e5005000)
    libcrypto.so.10 => /lib64/libcrypto.so.10 (0x00007f25e4ba4000)
    libunistring.so.2 => /lib64/libunistring.so.2 (0x00007f25e4874000)
    libkrb5support.so.0 => /lib64/libkrb5support.so.0 (0x00007f25e4664000)
    libkeyutils.so.1 => /lib64/libkeyutils.so.1 (0x00007f25e4460000)
    libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f25e4246000)
    libsasl2.so.3 => /lib64/libsasl2.so.3 (0x00007f25e4028000)
    libaesgm.so.0 => /lib64/libaesgm.so.0 (0x00007f25e3e16000)
    libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007f25e3bdf000)
    libfreebl3.so => /lib64/libfreebl3.so (0x00007f25e39dc000)
```

Without "%global _hardened_build 1" and manually fixing include and library paths for assimp.pc:
```
$ cat /usr/lib64/pkgconfig/assimp.pc
prefix=/usr
exec_prefix=/usr/bin
libdir=/usr//usr/lib64
includedir=/usr//usr/include/assimp

Name: Assimp
Description: Import various well-known 3D model formats in an uniform manner.
Version: 3.2.0
Libs: -L${libdir} -lassimp
Libs.private: -lstdc++ -lz
Cflags: -I${includedir}
```
```
$ sed -i '.bak' -e 's|usr//||' /usr/lib64/pkgconfig/assimp.pc
```
```
$ cat /usr/lib64/pkgconfig/assimp.pc
prefix=/usr
exec_prefix=/usr/bin
libdir=/usr/lib64
includedir=/usr/include/assimp

Name: Assimp
Description: Import various well-known 3D model formats in an uniform manner.
Version: 3.2.0
Libs: -L${libdir} -lassimp
Libs.private: -lstdc++ -lz
Cflags: -I${includedir}
```
It just works (i.e. without changes for pioneer.spec file), with following excerpt from config.log file, related to ASSIMP configuration:
```
configure:5872: checking for ASSIMP version >= 3.2
configure:5907: g++ -o conftest -I/usr/include  -Wl,-z,relro  conftest.cpp -lassimp  -ldl -lrt -lz -lassimp >&5
configure:5907: $? = 0
configure:5907: ./conftest
configure:5907: $? = 0
configure:5908: result: yes
```

By the way, the issue with assimp.pc file on Fedora is follows:
The Fedora's maintainer (for assimp)  [removed](http://pkgs.fedoraproject.org/cgit/rpms/assimp.git/tree/assimp-3.2.0-cmakefix.patch?id=c27ec6b408b700a30fd5a520a60f4bdad8467e52#n21) ASSIMP_ROOT_DIR (defined as CMAKE_INSTALL_PREFIX) from [assimp-config.cmake.in](https://github.com/assimp/assimp/blob/c318d648ff73aa2e32b783709d69740ac3e8c072/assimp-config.cmake.in#L46), but didn't do this for [assimp.pc.in](https://github.com/assimp/assimp/blob/c318d648ff73aa2e32b783709d69740ac3e8c072/assimp.pc.in) file:
```
--- assimp.pc.in.orig	2015-11-03 18:47:37 UTC
+++ assimp.pc.in
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=@CMAKE_INSTALL_PREFIX@/@ASSIMP_BIN_INSTALL_DIR@
-libdir=@CMAKE_INSTALL_PREFIX@/@ASSIMP_LIB_INSTALL_DIR@
-includedir=@CMAKE_INSTALL_PREFIX@/@ASSIMP_INCLUDE_INSTALL_DIR@/assimp
+libdir=@ASSIMP_LIB_INSTALL_DIR@
+includedir=@ASSIMP_INCLUDE_INSTALL_DIR@/assimp
 
 Name: @CMAKE_PROJECT_NAME@
 Description: Import various well-known 3D model formats in an uniform manner.
```
because of [assimp configuration](http://pkgs.fedoraproject.org/cgit/rpms/assimp.git/tree/assimp.spec?id=c27ec6b408b700a30fd5a520a60f4bdad8467e52#n102).

Maybe, there is a need to add "-fPIC" for [Cflags](https://github.com/assimp/assimp/blob/c318d648ff73aa2e32b783709d69740ac3e8c072/assimp.pc.in#L11) variable of assimp.pc file, when hardened build of assimp used on Fedora, because even minimal C/C++ program (about version check, as config.log showed) doesn't build without it.

--- a/configure.ac
+++ b/configure.ac
@@ -284,7 +284,7 @@ if test "x$cross_compiling" = "xno" ; then
 	assimp_rpath=`echo $ASSIMP_LIBS | sed -e 's/-[^L] *[^ ]\+ *//g' | sed -e 's/-L */-Wl,-rpath,/g'`
 	old_cxxflags="$CXXFLAGS"
 	old_libs="$LIBS"
-	CXXFLAGS="$ASSIMP_CFLAGS"
+	CXXFLAGS="$ASSIMP_CFLAGS $EXTRA_CXXFLAGS"
 	LIBS="$ASSIMP_LIBS $EXTRA_LIBS $assimp_rpath"
 	AC_LANG_PUSH([C++])
 	AC_RUN_IFELSE([AC_LANG_PROGRAM([[#include <assimp/defs.h>

